factory/dataComponents_test.go

```diff
@@ -0,0 +1,401 @@
+	require.Equal(t, factory.ErrNilEconomicsData, err)
+}
+
+func TestNewDataComponentsFactory_NilShardCoordinatorShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	args.ShardCoordinator = nil
+
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.Nil(t, dcf)
+	require.Equal(t, factory.ErrNilShardCoordinator, err)
+}
+
+func TestNewDataComponentsFactory_NilCoreComponentsShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	args.Core = nil
+
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.Nil(t, dcf)
+	require.Equal(t, factory.ErrNilCoreComponents, err)
+}
+
+func TestNewDataComponentsFactory_NilPathManagerShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	args.PathManager = nil
+
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.Nil(t, dcf)
+	require.Equal(t, factory.ErrNilPathManager, err)
+}
+
+func TestNewDataComponentsFactory_NilEpochStartNotifierShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	args.EpochStartNotifier = nil
+
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.Nil(t, dcf)
+	require.Equal(t, factory.ErrNilEpochStartNotifier, err)
+}
+
+func TestNewDataComponentsFactory_OkValsShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.NoError(t, err)
+	require.NotNil(t, dcf)
+}
+
+func TestDataComponentsFactory_CreateShouldErrDueBadConfig(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	args.Config.ShardHdrNonceHashStorage = config.StorageConfig{}
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.NoError(t, err)
+
+	dc, err := dcf.Create()
+	require.Error(t, err)
+	require.Nil(t, dc)
+}
+
+func TestDataComponentsFactory_CreateForShardShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	dcf, err := factory.NewDataComponentsFactory(args)
+
+	require.NoError(t, err)
+	dc, err := dcf.Create()
+	require.NoError(t, err)
+	require.NotNil(t, dc)
+}
+
+func TestDataComponentsFactory_CreateForMetaShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := getDataArgs()
+	multiShrdCoord := mock.NewMultiShardsCoordinatorMock(3)
+	multiShrdCoord.CurrentShard = core.MetachainShardId
+	args.ShardCoordinator = multiShrdCoord
+	dcf, err := factory.NewDataComponentsFactory(args)
+	require.NoError(t, err)
+	dc, err := dcf.Create()
+	require.NoError(t, err)
+	require.NotNil(t, dc)
+}
+
+func getDataArgs() factory.DataComponentsFactoryArgs {
+	return factory.DataComponentsFactoryArgs{
+		Config:             getGeneralConfig(),
+		EconomicsData:      &economics.EconomicsData{},
+		ShardCoordinator:   mock.NewOneShardCoordinatorMock(),
+		Core:               getCoreComponents(),
+		PathManager:        &mock.PathManagerStub{},
+		EpochStartNotifier: &mock.EpochStartNotifierStub{},
+		CurrentEpoch:       0,
+	}
+}
+
+func getGeneralConfig() *config.Config {
+	return &config.Config{
+		GeneralSettings: config.GeneralSettingsConfig{
```

Comment:
can we set instead only the relevant configuration here?

Reply:
Removed not relevant configuration. regarding the function that creates a test config, I don't think that valid storage configuration can be generated

Reply:
What that has to do with having a function that creates a config instance rather than creating the config all over the tests? What do you mean by "valid storage configuration"?

Reply:
I misunderstood. its is refactored now